File: python/recursion/lastIndex.py
#get last index of an element in a list
def lastIndex(arr,x):
    global c
    c-=1
    if(len(arr)>0):
        if(arr[len(arr)-1]==x):
            print(c)
            return
        arr.pop(len(arr)-1)
        lastIndex(arr,x)
    else:
        print(-1)


# Main
from sys import setrecursionlimit
setrecursionlimit(11000)
n=int(input())
arr=list(int(i) for i in input().strip().split(' '))
x=int(input())
c=len(arr)

lastIndex(arr,x)

File: javascript/recursion/lastIndex.js
//get last index of an element using recursion
let arr=[9 ,8 ,10 ,8];
let x=8;
let c=arr.length;
findLast(arr,x);

function findLast(arr,x){
    c--;
   if(arr.length>0){
        if(arr[arr.length-1]===x){
            console.log(c);
            return
        }
       arr.pop();

       findLast(arr,x);
   }else{
       console.log(-1);
   }

}
